ArtOfIllusion/src/artofillusion/MoveViewTool.java

```diff
@@ -118,8 +118,9 @@ public void mouseDragged(WidgetMouseEvent e, ViewerCanvas view)
 		if (view.getBoundCamera() != null)
 			view.getBoundCamera().getCoords().copyCoords(view.getCamera().getCameraCoordinates());
 		view.frustumShape.update();
-		if (ArtOfIllusion.getPreferences().getDrawActiveFrustum() || 
-		   (ArtOfIllusion.getPreferences().getDrawCameraFrustum() && view.getBoundCamera() != null))
+		if (theWindow != null &&
+		   (ArtOfIllusion.getPreferences().getDrawActiveFrustum() || 
```

Comment:
Querying for preferences value on every mouse move looks not the good solution. IMO implementing property change listener for propertie @ Preferences and listen for these changes @ tool class is more elegant.

Reply:
I'll agree, that this does not look very good. 

The reason to check the preferences is to avoid constant redraw on all views, when there is nothing changing.  Unfortunately if there are any moving graphics on the view, the only way to update them is the redraw the 3D-objects as well. 

Putting a listener here would be a bit of an overkill to me: It is not really possible, that the values of preferences would change during view rotation/move/zoom.  It would probably work best to check all the above conditions at mousePress, put the result into a `boolean updateWindow` and use that during drag. -- Did not come to my mind at the time, though.

I'm hoping to get rid of the problem entirely in the future by making the ViewerCanvas layered: Have the 3D on its dedicated (OGL)layer and do the mere graphics on a transparent layer on top of it. This is actually a necessary development step to modernize the graphic feedback to the user. Thinking of adding pre-hilight of objects / lines / vertices on mouseMove, implementing a real hilight of moving objects instead of just showing the drag handles, displaying warnings and cues directly on the view instead of popping up dialogs (which currently does not happen at all).

Reply:
> Putting a listener here would be a bit of an overkill to me: It is not really possible, that the values of preferences would change during view rotation/move/zoom
I mean that the fields in Preferences may be listenable and the XXXTool may listen on these changes
I try this approach in #47

Reply:
Either way, you then still need to store the preference value in a local variable.

Discussions like this need to be in their own issues - this is off topic for this PR, and will be harder to find later. (PR is about to be closed)